src/infra/flow-system-inmem/tests/tests/test_flow_service_inmem.rs

```diff
@@ -414,6 +432,237 @@ async fn test_manual_trigger() {
+
+    // Remember start time
+    let start_time = harness
+        .now_datetime()
+        .duration_round(Duration::try_milliseconds(SCHEDULING_ALIGNMENT_MS).unwrap())
+        .unwrap();
+
+    // Run scheduler concurrently with manual triggers script
+    tokio::select! {
+        // Run API service
+        res = harness.flow_service.run(start_time) => res.int_err(),
+
+        // Run simulation script and task drivers
+        _ = async {
+                  // Task 0: "foo" start running at 10ms, finish at 20ms
+                  let task0_driver = harness.task_driver(TaskDriverArgs {
+                    task_id: TaskID::new(0),
+                    dataset_id: Some(foo_id.clone()),
+                    run_since_start: Duration::try_milliseconds(10).unwrap(),
+                    finish_in_with: Some((Duration::try_milliseconds(10).unwrap(), TaskOutcome::Success(TaskResult::Empty))),
+                    expected_logical_plan: LogicalPlan::HardCompactDataset(HardCompactDataset {
+                      dataset_id: foo_id.clone(),
+                      max_slice_size: None,
+                      max_slice_records: None,
+                    }),
+                });
+                let task0_handle = task0_driver.run();
+
+                let task2_driver = harness.task_driver(TaskDriverArgs {
+                  task_id: TaskID::new(1),
+                  dataset_id: Some(bar_id.clone()),
+                  run_since_start: Duration::try_milliseconds(60).unwrap(),
+                  finish_in_with: Some((Duration::try_milliseconds(10).unwrap(), TaskOutcome::Success(TaskResult::Empty))),
+                  expected_logical_plan: LogicalPlan::HardCompactDataset(HardCompactDataset {
+                    dataset_id: bar_id.clone(),
+                    max_slice_size: None,
+                    max_slice_records: None,
+                  }),
+                });
+                let task2_handle = task2_driver.run();
+
+                // Manual trigger for "foo" at 00ms
+                let trigger0_driver = harness.manual_flow_trigger_driver(ManualFlowTriggerArgs {
+                    flow_key: foo_flow_key,
+                    run_since_start: Duration::try_milliseconds(10).unwrap(),
+                });
+                let trigger0_handle = trigger0_driver.run();
+
+                // Manual trigger for "bar" at 40ms
+                let trigger1_driver = harness.manual_flow_trigger_driver(ManualFlowTriggerArgs {
+                    flow_key: bar_flow_key,
+                    run_since_start: Duration::try_milliseconds(40).unwrap(),
+                });
+                let trigger1_handle = trigger1_driver.run();
+
+                // Main simulation script
+                let main_handle = async {
+                    // Moment 10ms - manual foo trigger happens here:
+                    //  - flow 0 gets trigger and is rescheduled to 40ms (20ms finish + 20ms throttling <= 40ms now)
```

Comment:
This comment is definitely copypasted. Nothing gets rescheduled for compacting flows